Framework/API/src/MatrixWorkspace.cpp

```diff
@@ -2016,6 +2035,12 @@ void MatrixWorkspace::updateCachedDetectorGrouping(const size_t index) const {
 void MatrixWorkspace::buildDefaultSpectrumDefinitions() {
   const auto &detInfo = detectorInfo();
   size_t numberOfDetectors{detInfo.size()};
+  if (numberOfDetectors == 0) {
+    // Default to empty spectrum definitions if there is no instrument.
+    m_indexInfo->setSpectrumDefinitions(
+        std::vector<SpectrumDefinition>(m_indexInfo->size()));
+    return;
+  }
```

Comment:
This is not related to MPI, it simply makes creating workspaces with `IndexInfo` simpler. Previously we were forced to manually set a vector of empty `SpectrumDefinition`, otherwise the workspaces would reject the `IndexInfo`.

My new policity here is: We build defaul spectrum definitions *whenever there is a unique way to do this*, i.e., 1:1 mapping, or all empty. Can anyone think of a reason why this would be bad?